### DISCUSSION TITLE
Redraw image when file contents changes

### DISCUSSION THREAD

**Discussion by cofeek-codes** [3rd/image.nvim#211] [Q&A] [answered]
i am working with plantuml and other "code-to-graphic" tools. I want to create an autocmd that will redraw image every time file changes.
So far i got this, but it draws old image, and i kinda don't know how to make it work
```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = "image_nvim",
  callback = function()
    vim.keymap.set("n", "r", function()
      local image = require("image").from_file(vim.fn.expand("%"))
      if image then
        image:clear()
        image:render()
      end
      print("image should reload")
    end, { silent = true, buffer = true })
  end,
})

```

**Comment by 3rd** [comment, marked as answer]
Hey, can you try this branch?  feat-handle-source-image-external-changes
Had some discussion in https://github.com/3rd/image.nvim/pull/150 about this and I think we got the right solution, but didn't merge it yet.

**Reply by cofeek-codes** [reply]
Works fine. Thanks! A question: is it necessary to put cursor on image? can it be done without it?

**Reply by 3rd** [reply]
It won't be necessary, but a rerender needs to be triggered. You can do it with a keybind / it could be a different event.
I'll get the change into master, don't stay on that branch.
But what's your PlantUML flow? I can add support for it to [diagram.nvim](https://github.com/3rd/diagram.nvim/) if you use it inside a type of document format.

**Reply by cofeek-codes** [reply]
I didn't get what you mean by flow, but i just open plantuml file and edit it and see the output in other pane. but about `diagram.nvim` yes, i'd like to have this thing in markdown files as well
![image](https://github.com/user-attachments/assets/6968a40f-900b-4809-a513-c9350e21443d)

**Reply by 3rd** [reply]
https://github.com/user-attachments/assets/a330b84e-29d6-41b6-a10e-7f4074edc9c3

**Reply by cofeek-codes** [reply]
Dosen't work for me(

![image](https://github.com/user-attachments/assets/49dd4600-0062-45dc-a2e1-258ab77e889d)

i brought it to my config with default options 
```lua
return {
  {
    "3rd/diagram.nvim",
    dependencies = {
      "3rd/image.nvim",
    },
    opts = {},
  },
}

```
I ensured that i have plantuml in my PATH
`whereis plantuml`: `plantuml: /usr/bin/plantuml /usr/share/man/man1/plantuml.1.gz`

**Reply by 3rd** [reply]
Needs to be in a plantuml code block.
  ```plantuml

**Reply by cofeek-codes** [reply]
this block of code is empty
![image](https://github.com/user-attachments/assets/c55eb51e-fbfa-4193-8b9f-1226c1efb7ed)

**Reply by 3rd** [reply]
![image](https://github.com/user-attachments/assets/c7c1f9ad-810c-443c-8c80-38cf9a437822)
like this

**Reply by 3rd** [reply]
![image](https://github.com/user-attachments/assets/dc1576fe-7c53-411c-a4c0-18cb58324605)
with the triple tick notation

**Reply by cofeek-codes** [reply]
Yay! it worked after i pasted the full version of default config in my file.
```lua
return {
  {
    "3rd/diagram.nvim",
    dependencies = {
      "3rd/image.nvim",
    },
    opts = { -- you can just pass {}, defaults below
      renderer_options = {
        mermaid = {
          background = nil, -- nil | "transparent" | "white" | "#hex"
          theme = nil, -- nil | "default" | "dark" | "forest" | "neutral"
          scale = 1, -- nil | 1 (default) | 2  | 3 | ...
        },
        plantuml = {
          charset = nil,
        },
      },
    },
  },
}

```
![image](https://github.com/user-attachments/assets/53b5f5dc-f659-4fdf-ac79-f16f8a1753f7)
Once again thank you so much for your continuous support.
I have a few more questions, but i think it is better to place them in diagram.nvim repo

**Reply by 3rd** [reply]
Yes, it must be a bug if it doesn't work with opts={}, will check!
